sstables/generation_type.hh

```diff
@@ -51,6 +51,14 @@ public:
         }
         return _value.get_least_significant_bits();
     }
+    /// tell if a generation identifier is supposed to be handled by the current shard
+    ///
+    /// the integer based generation is used as a hint to tell if the SSTable with
+    /// this generation should be handled by a given shard. but this does not applies
+    /// UUID based generation.
```

Comment:
We don't know that, we may yet find out a way.

I would replace these two sentences with just: "This method only returns a hint, whether an sstable belongs to a shard is determined by overlapping its partition-ranges with the shard's owned ranges.

Reply:
Agreed

Reply:
thanks. incorporated.